File: gameserver/src/main/java/l2s/gameserver/network/l2/s2c/ExVariationCancelResult.java
package l2s.gameserver.network.l2.s2c;

import l2s.gameserver.network.l2.OutgoingExPackets;

public class ExVariationCancelResult implements IClientOutgoingPacket
{
	private int _closeWindow;
	private int _unk1;

	public ExVariationCancelResult(int result)
	{
		_closeWindow = 1;
		_unk1 = result;
	}

	@Override
	public boolean write(l2s.commons.network.PacketWriter packetWriter)
	{
		OutgoingExPackets.EX_VARIATION_CANCEL_RESULT.writeId(packetWriter);
		packetWriter.writeD(_unk1);
		packetWriter.writeD(_closeWindow);

		return true;
	}
}

File: gameserver/src/main/java/l2s/gameserver/network/l2/s2c/augmentation/ExVariationCancelResult.kt
package l2s.gameserver.network.l2.s2c.augmentation

import l2s.gameserver.network.l2.s2c.L2GameServerPacket

class ExVariationCancelResult(private val openWindow: Boolean, private val result: Boolean) : L2GameServerPacket() {

    override fun writeImpl() {
        writeD(result)
        writeD(openWindow)
    }

    companion object {
        val CLOSE: L2GameServerPacket = ExVariationCancelResult(false, false)
        val FAIL: L2GameServerPacket = ExVariationCancelResult(true, false)
        val SUCCESS: L2GameServerPacket = ExVariationCancelResult(true, true)
    }

}